server/src/test/java/org/elasticsearch/common/geo/GeoWKTShapeParserTests.java

```diff
@@ -285,7 +284,8 @@ public void testParseMixedDimensionPolyWithHole() throws IOException {
             .put(IndexMetaData.SETTING_INDEX_UUID, UUIDs.randomBase64UUID()).build();
 
         Mapper.BuilderContext mockBuilderContext = new Mapper.BuilderContext(indexSettings, new ContentPath());
-        final GeoShapeFieldMapper mapperBuilder = new GeoShapeFieldMapper.Builder("test").ignoreZValue(false).build(mockBuilderContext);
+        final LegacyGeoShapeFieldMapper mapperBuilder =
+            (LegacyGeoShapeFieldMapper)(new LegacyGeoShapeFieldMapper.Builder("test").ignoreZValue(false).build(mockBuilderContext));
```

Comment:
should we test with the non-legacy mapper instead?

Reply:
:+1: